File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        hm = {}
        temp = ListNode(-1)
        temp.next = head
        ans = temp
        i =0
        while temp:
            hm[i]=temp
            temp = temp.next
            i+=1
        if n==1:
            x = i-n
            hm[x-1].next = None
        else:
            j = i-n
            p = hm[j-1]
            n = hm[j+1] if j+1<i else None
            p.next = n
        return ans.next
            
            
        

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getLength(ListNode node){
        int ans = 0;
        ListNode temp = node;
        while(temp!=null){
            ans+=1;
            temp = temp.next;
        }
        return ans;
    }
    
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode temp = head;
        int l = getLength(head);
        int x = l-n-1;
        while(x>0){
            temp = temp.next;
            x-=1;
        }
        
        if(x<0){
            return head.next;
        }
        
        if(temp.next!=null){
             temp.next=temp.next.next;
        }
        return head;
    }
}